live_data/websocket_subscriber/live_data_consumer.py

```diff
@@ -0,0 +1,66 @@
+import os
+import logging
+import traceback
+import threading
+import asyncio
+
+from concurrent.futures.thread import ThreadPoolExecutor
+
+from d3a_api_client.utils import consumer_websocket_domain_name_from_env, domain_name_from_env, \
+    retrieve_jwt_key_from_server, RestCommunicationMixin
+from d3a_api_client.websocket_device import WebsocketMessageReceiver
+from d3a_api_client.constants import MAX_WORKER_THREADS
+from d3a_api_client.websocket_device import retry_coroutine
+from d3a_api_client.rest_device import RestDeviceClient
+
+
+class LiveDataWebsocketMessageReceiver(WebsocketMessageReceiver):
+
+    def received_message(self, message):
+        try:
+            if "event" in message and message["event"] == "live_data":
+                for api_args in self.client.device_api_client_mapping[message['data']['area_name']]:
+                    RestDeviceClient(**api_args).set_energy_forecast(
```

Comment:
Instead of constructing a new RestDeviceClient object for all messages, you can reuse the `self.client` IIRC

Reply:
After offline discussion with @faizan2590 , this is not feasible because the same websocket is used for multiple devices, therefore the rest client will have to know to which device to send this measurement. Should remain as is.